### DISCUSSION TITLE
A Thank You note

### DISCUSSION THREAD

**Discussion by 4yerba** [TG9541/stm8ef#451] [General]
I've just watched your presentation of the ModBus stm8ef example on the forth2020 YT channel. For a second time. With even more interest and pleasure and thought I'd stop by to say "thank you". I was really surprised there were no questions. Have I been a FB user and a member of this group attending the meeting there would sure be at least one question from me.

Again, thanks for the excellent presentation and especially for all of the work that led you to it.

**Comment by TG9541** [comment]
@4yerba thanks for the nice words! 

It always takes more time to make a demo (and to create a viable video set-up) than planned. The way I do things is often a bit out of the ordinary and this solution uses special STM8 eForth properties. Peter once asked me if it can be ported to other eForth implementation but this question can't be answered from the Forth point-of view - it's the design of the Forth system (e.g., idle task, serial interface interrupt) that counts. I guess that the audience simply didn't know where to start.

There are a few people who actually use stm8ef-modbus and it's not uncommon that I receive questions from them :-)

**Reply by 4yerba** [reply]
Yes, that is true. The more clever the software implementation that makes use of the lowest level hardware features in the most efficient way the harder it is to port it to another platform. Without a common hardware platform one needs several abstraction layers to try and make high level programs more universal and portable but that is not the nature of Forth as I understand it. At least not on tiny microcontrollers where every byte of memory and clock cycle counts. Any attempt that aims at making tons of universal, reusable programs or libraries will eventually fail. I think people who actually use Forth are the ones who like to do things their own way and it will be really hard to persuade them to give up this freedom. Common problems can be solved with common tools. Unique problems can be solved easier with Forth, one by one, custom tailored. I may, but do not want to, be wrong here :) .
